packages/docs-components/src/knobs/KnobInspector.tsx

```diff
@@ -46,30 +47,37 @@ const getKnobComponents = (
   return components
 }
 
-const KnobInspector: React.FunctionComponent = () => {
+type KnobInspectorProps = {
+  children?: (children: React.ReactElement) => React.ReactElement
+}
+
+const KnobInspector: React.FunctionComponent<KnobInspectorProps> = props => {
   const knobsContext = React.useContext(KnobsContext)
 
   const { Control, Field, Label } = getKnobControls(knobsContext)
   const knobComponents = getKnobComponents(knobsContext)
   const knobValues = useKnobValues()
 
-  return (
-    <>
-      {knobValues.map((knob: KnobDefinition) => {
-        const setValue = (value: any) => knobsContext.setKnobValue(knob.name, value)
-        const knobProps: KnobComponentProps = { ...knob, setValue }
+  const children =
+    knobValues.length > 0 ? (
+      <>
+        {knobValues.map((knob: KnobDefinition) => {
+          const setValue = (value: any) => knobsContext.setKnobValue(knob.name, value)
+          const knobProps: KnobComponentProps = { ...knob, setValue }
+
+          return (
+            <Field {...knobProps} key={knob.name}>
+              <Label {...knobProps} />
+              <Control {...knobProps}>
+                {React.createElement(knobComponents[knob.type], knobProps)}
+              </Control>
+            </Field>
+          )
+        })}
+      </>
+    ) : null
 
-        return (
-          <Field {...knobProps} key={knob.name}>
-            <Label {...knobProps} />
-            <Control {...knobProps}>
-              {React.createElement(knobComponents[knob.type], knobProps)}
-            </Control>
-          </Field>
-        )
-      })}
-    </>
-  )
+  return props.children ? props.children(children) : children
```

Comment:
`children()` confuses me. Can we use `render` prop?

Reply:
I want to follow React API, they use `children`: https://reactjs.org/docs/context.html#contextconsumer